### PR TITLE
Implement span interaction with Context.

### PR DESCRIPTION
**Pull request by bogdandrutu**

### CODE REVIEW

File: sdk/src/main/java/openconsensus/sdk/trace/internal/CurrentSpanUtils.java [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+   * object that represents that scope. The scope is exited when the returned object is closed.
+   *
+   * <p>Supports try-with-resource idiom.
+   *
+   * @param span The {@code Span} to be set to the current context.
+   * @param endSpan if {@code true} the returned {@code Scope} will close the {@code Span}.
+   * @return An object that defines a scope where the given {@code Span} is set to the current
+   *     context.
+   */
+  public static Scope withSpan(Span span, boolean endSpan) {
```
Comment (carlosalberto) [inline sdk/src/main/java/openconsensus/sdk/trace/internal/CurrentSpanUtils.java]: This method was removed from the API, and I'm wondering if we could remove it here as well.

Comment (carlosalberto) [reply sdk/src/main/java/openconsensus/sdk/trace/internal/CurrentSpanUtils.java]: (and same for overloads taking a `endSpan` param)

Comment (bogdandrutu) [reply sdk/src/main/java/openconsensus/sdk/trace/internal/CurrentSpanUtils.java]: I removed the endSpan part. Curious if you suggest to remove the "withSpan" overall.

File: sdk/src/main/java/openconsensus/sdk/trace/internal/CurrentSpanUtils.java [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+          span.end();
+        }
+      }
+    }
+  }
+
+  private static void setErrorStatus(Span span, Throwable t) {
+    span.setStatus(
+        Status.UNKNOWN.withDescription(
+            t.getMessage() == null ? t.getClass().getSimpleName() : t.getMessage()));
```
[the remark this answers is not in the document]
Comment (bogdandrutu) [reply sdk/src/main/java/openconsensus/sdk/trace/internal/CurrentSpanUtils.java]: Not yet, please file an issue for this.

**Pull request merged**